### PR TITLE
Support generating configuration files from consul

### PR DESCRIPTION
**Pull request by ianmcorvidae**
This is the service-side of generating configuration files from consul. See the ansible PR at https://github.com/cyverse-de/ansible/pull/5/files as well.

**My Open Questions**
1. <s>Would it make sense for me to make a new repository to store the `run-service` script? It's general enough to be used for everything, as far as I know (though thus far I've only tested with the UI and this). I'm not sure if we'd rather copy it between repos or depend on something external; I think it's on the line as far as complexity for that decision.</s>
2. <s>Do we want our own base images that do some of the stuff we'll be doing with everything? For this go stuff, it'd be `golang:1.7-alpine` plus the `apk add`, adding `go-junit-report`, installing `consul-template`, and adding `run-service`. Same basic idea for clojure, if with a different set of things to install (perhaps notably, the static config file which we use for clojars deploys would probably go in). If we do this, the repo(s) which make these images would probably be the shared location for the above question. Why I'm not sure about this is that it seems like it might be a lot to manage, especially if we want to devise some way to regularly pull in updates to the base images. But then, we're not pulling those updates in now, so...</s>
3. The current setup can in some cases produce a few warnings on startup which can be ignored (about missing keys in consul or problems from viper merging hashes). I don't know if we care enough to try to reliably silence those, though I'm leaning toward no.
4. Are the templates below readable enough for everyone?

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
Some other extra work not yet done: adding in the right stuff to the docker-compose files such that our actual deployment uses consul. But the images this PR generates can still be used in the way we currently use them, so it's not required right away.

**Comment by johnworth** [comment]
It kind of sounds like we should version-control run-service separately from the services and create new base images that include it. That would allow us to keep the service builds simple, rather than have to coordinate pulling files in from multiple repos for what should be a simple Docker build.

If we go that route, we might want to keep the run-service script in the same repo as the Dockerfiles for the new base-images. We could either create a new repo, or put run-service and the new base image Dockerfiles into the dockerfiles repo.

**Comment by ianmcorvidae** [comment]
That seems good to me, yeah. I've made a new repo `baseimages` for this, since I'd rather have more repos split by contexts. Right now it just builds a `discoenv/golang-base` image, but others can get added as appropriate.

File: Dockerfile [hunk 1]
```diff
@@ -1,15 +1,13 @@
...
+ENV CONF_TEMPLATE=/go/src/github.com/cyverse-de/jex-adapter/jobservices.yml.tmpl
+ENV CONF_FILENAME=jobservices.yml
+ENV PROGRAM=jex-adapter
 
 COPY . /go/src/github.com/cyverse-de/jex-adapter
 RUN go install github.com/cyverse-de/jex-adapter
 
 EXPOSE 60000
-ENTRYPOINT ["jex-adapter"]
-CMD ["--help"]
```
Comment (ianmcorvidae) [inline Dockerfile:12]: I've put the new entrypoint (`run-service`) and the default `CMD` into `golang-base`, since they're usually going to be the same.

**Comment by ianmcorvidae** [comment]
(merging this -- our current approach should continue to work, and merging this will help validate that)

**Pull request merged**